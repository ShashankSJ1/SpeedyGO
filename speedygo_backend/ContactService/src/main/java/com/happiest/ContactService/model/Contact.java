package com.happiest.ContactService.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "contact")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Contact {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long contactId;

    @Column
    private String email;

    @Column
    private String name;

    @Column
    private String message;
}
