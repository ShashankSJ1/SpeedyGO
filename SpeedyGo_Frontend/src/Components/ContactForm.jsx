import React from 'react';
import { useFormik } from 'formik';
import { FaMapMarkerAlt, FaPhoneAlt, FaEnvelope } from 'react-icons/fa';
import '../styles/Contact.css';
import { contactInitialValues, contactValidationSchema, CONTACT_INFO, CONTACT_UI_TEXT,CONTACT_ENDPOINT,CONTACT_ERROR } from '../constants';
import CustomFormInput from './CustomComponents/CustomFormInput';
import CustomButton from './CustomComponents/CustomButton';
import Cookies from 'js-cookie';
import axios from 'axios';
import Swal from 'sweetalert2';

const ContactForm = () => {
  const token = Cookies.get('token');
  const formik = useFormik({
    initialValues: contactInitialValues,
    validationSchema: contactValidationSchema,
    onSubmit: async (values) => {
      const data = {
        email:values.email,
        name:values.name,
        message:values.message
      }
      try {
        console.log(CONTACT_ENDPOINT)
        const response = await axios.post(CONTACT_ENDPOINT,data, {
          headers: {
            'Authorization': `Bearer ${token}`
          }
        })
        console.log(response);
        Swal.fire('Success', response.data, 'success')
      }
      catch(error){
        Swal.fire('Error',CONTACT_ERROR, 'error')
      }
    },
  });

  return (
    <div className="form-wrapper">
      <div className="contact-container">
        <div className="contact-content">
          <div className="left-side">
            <div className="address details">
              <FaMapMarkerAlt className="icon" />
              <div className="topic">{CONTACT_INFO.ADDRESS.TOPIC}</div>
              <div className="text-one">{CONTACT_INFO.ADDRESS.LINE1}</div>
              <div className="text-two">{CONTACT_INFO.ADDRESS.LINE2}</div>
            </div>
            <div className="phone details">
              <FaPhoneAlt className="icon" />
              <div className="topic">{CONTACT_INFO.PHONE.TOPIC}</div>
              <div className="text-one">{CONTACT_INFO.PHONE.NUMBER1}</div>
              <div className="text-two">{CONTACT_INFO.PHONE.NUMBER2}</div>
            </div>
            <div className="email details">
              <FaEnvelope className="icon" />
              <div className="topic">{CONTACT_INFO.EMAIL.TOPIC}</div>
              <div className="text-one">{CONTACT_INFO.EMAIL.EMAIL1}</div>
              <div className="text-two">{CONTACT_INFO.EMAIL.EMAIL2}</div>
            </div>
          </div>

          <div className="right-side">
            <div className="topic-text">{CONTACT_UI_TEXT.SEND_MESSAGE}</div>
            <p>{CONTACT_UI_TEXT.MESSAGE_PROMPT}</p>

            <form onSubmit={formik.handleSubmit}>
              <CustomFormInput
                tag="input"
                type="text"
                name="name"
                placeholder={CONTACT_UI_TEXT.NAME_PLACEHOLDER}
                label={CONTACT_UI_TEXT.NAME_LABEL}
                value={formik.values.name}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.name && formik.errors.name && (
                <div className="error">{formik.errors.name}</div>
              )}

              <CustomFormInput
                tag="input"
                type="email"
                name="email"
                placeholder={CONTACT_UI_TEXT.EMAIL_PLACEHOLDER}
                label={CONTACT_UI_TEXT.EMAIL_LABEL}
                value={formik.values.email}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
              />
              {formik.touched.email && formik.errors.email && (
                <div className="error">{formik.errors.email}</div>
              )}

              <CustomFormInput
                tag="textarea"
                name="message"
                placeholder={CONTACT_UI_TEXT.MESSAGE_PLACEHOLDER}
                label={CONTACT_UI_TEXT.MESSAGE_LABEL}
                value={formik.values.message}
                onChange={formik.handleChange}
                onBlur={formik.handleBlur}
                rows="5"
              />
              {formik.touched.message && formik.errors.message && (
                <div className="error">{formik.errors.message}</div>
              )}

              <CustomButton buttonTitle={CONTACT_UI_TEXT.SUBMIT_BUTTON} type="submit"/>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
};

export default ContactForm;
