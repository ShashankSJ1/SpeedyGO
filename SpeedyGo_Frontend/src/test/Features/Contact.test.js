// Contact.test.js
import React from 'react';
import { render } from '@testing-library/react';
import Contact from '../../pages/Contact'; // Adjust the import path as needed
import ContactForm from '../../Components/ContactForm';; // Adjust the import path as needed

jest.mock('../../Components/ContactForm', () => () => <div>Mocked Contact Form</div>);

describe('Contact Component', () => {
  it('should render the ContactForm component', () => {
    const { getByText } = render(<Contact />);
    
    // Check if ContactForm is rendered
    expect(getByText('Mocked Contact Form')).toBeInTheDocument();
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(<Contact />);
    
    // Snapshot test
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the correct structure', () => {
    const { container } = render(<Contact />);
    
    // Check if the main div is present
    expect(container.querySelector('div')).toBeInTheDocument();
    expect(container.firstChild).toBeInTheDocument();
  });
});
