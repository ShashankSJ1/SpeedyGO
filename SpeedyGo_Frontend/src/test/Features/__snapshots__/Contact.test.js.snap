// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Mocked Contact Form
    </div>
  </div>
</DocumentFragment>
`;
